xarray/core/dataset.py

```diff
@@ -2366,6 +2339,58 @@ def unstack(self, dim):
 
         return self._replace_vars_and_dims(variables, coord_names)
 
+    def unstack(self, dim=None):
+        """
+        Unstack existing dimensions corresponding to MultiIndexes into
+        multiple new dimensions.
+
+        New dimensions will be added at the end.
+
+        Parameters
+        ----------
+        dim : str or sequence of str, optional
+            Dimension(s) over which to unstack. By default unstacks all
+            MultiIndexes.
+
+        Returns
+        -------
+        unstacked : Dataset
+            Dataset with unstacked data.
+
+        See also
+        --------
+        Dataset.stack
+        """
+        dim_from_kwarg = dim is not None
+
+        if isinstance(dim, basestring):
+            dims = set([dim])
+        elif dim is None:
+            dims = set(self.dims)
+        else:
+            dims = set(dim)
+
+        missing_dims = [dim for dim in dims if dim not in self.dims]
+        if missing_dims:
+            raise ValueError('Dataset does not contain the dimensions: %s'
+                             % missing_dims)
+
+        non_multi_dims = [dim for dim in dims
+                          if not isinstance(self.get_index(dim), pd.MultiIndex)]
+        if non_multi_dims and dim_from_kwarg:
+            raise ValueError('cannot unstack dimensions that do not '
+                             'have a MultiIndex: %s' % non_multi_dims)
+
+        dims = dims - set(non_multi_dims)
+        if len(dims) == 0:
+            raise ValueError('cannot unstack an object that does not have '
+                             'MultiIndex dimensions')
```

Comment:
I think that we can allow to `unstack` an object without `MultiIndex`, which just returns as is.
It would be useful if users want to remove any MultiIndexes from an object.